packages/framework/src/client/scheduler.js

```diff
@@ -1,20 +1,31 @@
 import debugFactory from 'debug';
-import { find } from 'lodash';
+import { find, groupBy, isArray, isEmpty, isEqual, isNil, uniq } from 'lodash';
 import ResourceRequest, { STATUS } from './resource-request';
 
 const debug = debugFactory( 'fresh-data:scheduler' );
 
 export default class Scheduler {
+	/**
+	 * Creates a new Scheduler
+	 * @param {Object} operations The mapping of operation names to functions
+	 * @param {function} setTimeout The function to set a time (defaults to window.setTimeout)
+	 * @param {function} clearTimeout The function to clear a timer by id (defaults to window.setTimeout)
+	 */
 	constructor(
-		fetch,
+		operations,
 		setTimeout = window.setTimeout,
 		clearTimeout = window.clearTimeout,
 	) {
-		this.fetch = fetch;
+		this.debug = debugFactory( 'fresh-data:scheduler' );
+
+		this.operations = operations;
 		this.requests = [];
 		this.setTimeout = setTimeout;
 		this.clearTimeout = clearTimeout;
 		this.timeoutId = null;
+
+		this.dataRequested = () => this.debug( 'warning: dataRequested called before setDataHandlers' );
```

Comment:
Minor: I kept tripping over the names `dataRequested`, `dataReceived`, and `requested` (from `ResourceRequest`). Maybe something like `handleDataReceived` for `dataRequested`, etc, would help clarify that this calling a function?

Reply:
I'm considering internalizing the state for fresh-data and essentially removing redux altogether (as I don't think it's really needed anymore anyway). If/when I do that, this will change for sure. Is it okay to keep these names as they are for now?

Reply:
I think it's not a huge issue - got used to it eventually. Wanted to point it out just in case it seemed important to you as well.